vmdb/app/models/ems_refresh/save_inventory_container.rb

```diff
@@ -51,7 +51,16 @@ def save_container_groups_inventory(ems, hashes, target = nil)
                 []
               end
 
-    save_inventory_multi(:container_groups, ems, hashes, deletes, [:ems_ref], [:container_definitions, :containers, :labels])
+    hashes.each do |h|
+      next if h[:container_node_name].nil?
+      node = ContainerNode.first(
+        :conditions => {:ems_id => ems.id, :name => h[:container_node_name]})
+      h[:container_node_id] = node.nil? ? nil : node.id
```

Comment:
You determine the linking by name in the parser code, then put a hash reference directly in the Hash.  Typically this is simplified by accessing the already parsed value from the `@data_index`.

In the parser it would look something like (pseduocode)

```ruby
def process_nodes
  process_collection(:container_nodes, connection.nodes) { |n| parse_node(n) }

  # I'm assuming the unique ref for nodes is not name, so you might need a secondary index by name with something like the following:
  @data[:container_nodes].each { |n| @data_index.store_path(:container_nodes, :by_name, n.name, n) }
end

def process_pods
  process_collection(:container_groups, connection.pods) { |p| parse_pod(p) }
end

def parse_pod(pod)
  new_result = {
    # bunch of attributes
  }

  new_result[:container_node] = @data_index.fetch_path(:container_nodes, :by_name, pod.spec.host) if pod.spec.respond_to?(:host)

  # ...
end
```